### PR TITLE
Modernize database tests and cleanup

### PR DESCRIPTION
**Pull request by SUPERCILEX**
@samtstern Since it looks like #481 isn't going to be merged anytime soon, this PR gives @CodingDoug credit for his work while allowing us to move forward with modernized tests.

### CODE REVIEW

File: database/src/androidTest/java/com/firebase/ui/database/FirebaseArrayOfObjectsTest.java [hunk 1]
```diff
@@ -35,54 +33,46 @@
...
-    @Test
-    public void testSize() throws Exception {
-        assertEquals(3, mArray.getCount());
+        mArray.cleanup();
+        mRef.getRoot().removeValue();
     }
 
     @Test
     public void testPushIncreasesSize() throws Exception {
-        assertEquals(3, mArray.getCount());
```
Comment (samtstern) [inline database/src/androidTest/java/com/firebase/ui/database/FirebaseArrayOfObjectsTest.java]: Was dropping this line intentional?

Comment (SUPERCILEX) [inline database/src/androidTest/java/com/firebase/ui/database/FirebaseArrayOfObjectsTest.java]: @CodingDoug removed it in his PR. If you'd like I'll add it back (I'm not sure why he removed it).

**Comment by samtstern** [comment]
@CodingDoug any issues with merging in your test changes now since we all agree they are good?  You can then rebase your PR and it will only contain the `Map` performance improvements.

**Comment by CodingDoug** [comment]
@samtstern Yeah, that's fine.  I promise I'll get back to the rest of this PR some time!

File: database/src/androidTest/java/com/firebase/ui/database/FirebaseArrayOfObjectsTest.java (hunk 1 above)
Comment (samtstern) [inline database/src/androidTest/java/com/firebase/ui/database/FirebaseArrayOfObjectsTest.java]: It does seem a little strange to assert that the count after is "4" but not assert that it was "3" before.  4 is only correct if it's one more than the initial count.

Comment (CodingDoug) [inline database/src/androidTest/java/com/firebase/ui/database/FirebaseArrayOfObjectsTest.java]: This check is redundant.  The '@Before' annotated function gets run before every single '@Test' function, and it verifies the size of the array ahead of time.

Comment (SUPERCILEX) [inline database/src/androidTest/java/com/firebase/ui/database/FirebaseArrayOfObjectsTest.java]: @CodingDoug thanks for the explanation!

**Pull request merged**